lib/instance.js

```diff
@@ -817,18 +817,23 @@ Instance.prototype.destroy = function(options) {
       return this.Model.runHooks('beforeDestroy', this, options);
     }
   }).then(function() {
-    var where;
+    var primaryKeys = this.Model.primaryKeyAttributes;
+    var where = {};
+
+    for (var i = 0; i < primaryKeys.length; i++) {
+      where[this.Model.rawAttributes[primaryKeys[i]].field] = this.get(primaryKeys[i], { raw: true });
+    }
```

Comment:
Good idea. I had to fix a small bug in the `where` function though. Please take a look.